### ISSUE TITLE
mixlib-shellout regression on windows in 12.6.0

### ISSUE THREAD

**Issue by bradwilliamson** [chef/chef#4464]
### Version:
[Chef-Client 12.6.0]

### Scenario:
With Chef-Client 12.5.1 I could use Ruby_block with Module Chef::Mixin::ShellOut to pass username and password
to gain access to windows shares, Copy files Down etc. I also used this as a method to kick off an install of SQL Server 2012 with an ISO mounted as Local Administrator on the remote server through my chef-workstation over WinRM. Now with Chef-Client 12.6.0 I get an error message stating it was expecting a boolean parameter & kills the chef-client process. 

### Steps to Reproduce:
[Use Ruby_block with Module: Chef::Mixin::ShellOut to try to launch a remote application or install with :username => and :password => . ]

### Expected Result:
[Hopefully fix the issue]

### Actual Result:
[I get the same error over and over now unless I use 12.5.1 client]

**Comment by thommay** [comment]
ping @chef/client-windows

**Comment by lamont-granquist** [comment]
Can you copy and paste the actual code and the error you get on the run, along with the stacktrace.out?

**Comment by bradwilliamson** [comment]
I'll have those today. Starting to work on collecting information requested.

**Comment by bradwilliamson** [comment]
[chef-stacktrace.txt](https://github.com/chef/chef/files/107399/chef-stacktrace.txt)

code


[rubyblocksqlinstall.txt](https://github.com/chef/chef/files/107401/rubyblocksqlinstall.txt)

**Comment by lamont-granquist** [comment]
```
Generated at 2016-01-27 15:40:41 -0600
TypeError: ruby_block[Chef is now installing SQL Server] (vsphere::rubyblocksqlinstall line 2) had an error: TypeError: wrong argument type  (expected a boolean parameter)
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/mixlib-shellout-2.2.5-universal-mingw32/lib/mixlib/shellout/windows/core_ext.rb:293:in `CreateProcessAsUserW'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/mixlib-shellout-2.2.5-universal-mingw32/lib/mixlib/shellout/windows/core_ext.rb:293:in `create'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/mixlib-shellout-2.2.5-universal-mingw32/lib/mixlib/shellout/windows.rb:80:in `run_command'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/mixlib-shellout-2.2.5-universal-mingw32/lib/mixlib/shellout.rb:259:in `run_command'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/mixin/shell_out.rb:97:in `shell_out_command'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/mixin/shell_out.rb:50:in `shell_out'
C:/chef/cache/cookbooks/vsphere/recipes/rubyblocksqlinstall.rb:6:in `block (2 levels) in from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/provider/ruby_block.rb:35:in `call'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/provider/ruby_block.rb:35:in `block in action_run'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/mixin/why_run.rb:52:in `call'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/mixin/why_run.rb:52:in `add_action'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/provider.rb:175:in `converge_by'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/provider/ruby_block.rb:34:in `action_run'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/provider.rb:144:in `run_action'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/resource.rb:596:in `run_action'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/runner.rb:74:in `run_action'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/runner.rb:106:in `block (2 levels) in converge'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/runner.rb:106:in `each'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/runner.rb:106:in `block in converge'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/runner.rb:105:in `converge'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/client.rb:658:in `block in converge'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/client.rb:653:in `catch'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/client.rb:653:in `converge'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/client.rb:692:in `converge_and_save'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/client.rb:271:in `run'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/application.rb:243:in `run_with_graceful_exit_option'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/application.rb:220:in `block in run_chef_client'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/local_mode.rb:44:in `with_server_connectivity'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/application.rb:203:in `run_chef_client'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/application/client.rb:413:in `block in interval_run_chef_client'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/application/client.rb:403:in `loop'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/application/client.rb:403:in `interval_run_chef_client'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/application/client.rb:393:in `run_application'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/lib/chef/application.rb:58:in `run'
C:/opscode/chef/embedded/lib/ruby/gems/2.0.0/gems/chef-12.6.0-universal-mingw32/bin/chef-client:26:in `<top (required)>'
C:/opscode/chef/bin/chef-client:64:in `load'
C:/opscode/chef/bin/chef-client:64:in `<main>'%
```

**Comment by lamont-granquist** [comment]
So smells a bit like a bug in the windows side of mixlib-shellout

**Comment by jaym** [comment]
@ksubrama might have run into this before. I think bugs were getting fixed in our dependencies.

**Comment by mwrock** [comment]
I started troubleshooting last night but fell asleep. I was able to reproduce when run via TK but not locally. I'll dig deeper today.

**Comment by mwrock** [comment]
here is my test recipe:
```
ruby_block 'testing' do
  block do
    Chef::Resource::RubyBlock.send(:include, Chef::Mixin::ShellOut)
    command_to_run = "notepad.exe"
    shell_out(command_to_run,
      {
        :user   => 'vagrant',
        :password => 'vagrant'
      }
    )
    action :run
  end
end
```

**Comment by bradwilliamson** [comment]
I took the single quotes out of the command_to_run and still got the error. Tried adding a , after the password value as well.  Here is what is in the Application Logs in Event viewer on the SQL Server that was provisioned. I


[applog.txt](https://github.com/chef/chef/files/109008/applog.txt)

**Comment by mwrock** [comment]
@bradwilliamson How are you invoking the chef run? I'm able to reproduce this error remotely (TK) but it works if i run locally with `chef-client -z` or `chef-apply`. I just want to make sure this matches what you are seeing.

I am guessing there is a regression in some impersonation code around shellout that does not play nice in a remote context. Will digging deeper soon.

**Comment by bradwilliamson** [comment]
I am running it remotely with chef-client -o 'vsphere::newbatch1' -c C:\chef-repo\.chef\knife.rb on my Windows Chef Workstation with ChefDK 0.10.0. When I pin it to chef-client 12.5.1 in the convergence options it works fine while provisioning it or running it remotely. Seems to be isolated to 12.6.0 as of now. I didn't have it pinned to a specific chef-client version in convergence options & chef-provisioning auto updated it to 12.6.0 is how I discovered it.

**Comment by mwrock** [comment]
this is addressed by https://github.com/chef/mixlib-shellout/pull/118. Closing here.

**Comment by mwrock** [comment]
FYI: released fix in mixlib-shellout 2.2.6 https://rubygems.org/gems/mixlib-shellout/versions/2.2.6-universal-mingw32
